lldb/source/Breakpoint/BreakpointOptions.cpp

```diff
@@ -534,7 +534,8 @@ void BreakpointOptions::GetDescription(Stream *s,
 
     if (m_ignore_count > 0)
       s->Printf("ignore: %d ", m_ignore_count);
-    s->Printf("%sabled ", m_enabled ? "en" : "dis");
+    s->PutCStringColorHighlighted(m_enabled ? "enabled " : "disabled ",
+                                  m_disbaled_breakpoint_highlight_settings);
```

Comment:
This is going to regex search for disabled and then replace it with its colored variant. That seems unnecessarily inefficient. You know whether you're going to put enabled or disabled, so you could print it colored right away. 

You might also have to check the debugger setting for whether colors are enabled (`Debugger::GetUseColor()`). Sometimes it's set at the stream level: the stream might have colors disabled if it was created somewhere were the setting was checked). I don't know if that's always going to be the case here. If it's not, we should check the setting here. To find out I would run lldb with `--no-use-colors` and see if the colors still get printed.

Reply:
Ah, so using `Printf` with `ansi::FormatAnsiTerminalCodes` instead? And yes using this with `--no-use-colors` does show the colour here so it looks like this isn't being checked at the stream level.

Reply:
Note, you can get this output both when lldb decides to print in the terminal, and when someone does:

```
stream = lldb.SBStream()
bkpt.GetDescription(stream, eDescriptionLevelWhatever)

```

At present SBStream's are backed by StreamString objects which get constructed with use_color -> false.  So you should be okay there.

Reply:
It seems odd you can't ask the stream whether it is using colors or not.  Because if I'm running with use-colors on, but I want to use the description of a breakpoint w/o colors - say to put it in a log - then I would correctly make an SBStream which is supposed to not use colors, and pass that to the description.  If you ONLY check whether the debugger is using colors, and override the stream's setting, then you will do the wrong thing.

Reply:
This means that only the debugger tells you whether or not you're using colours then? `BreakpointOptions::GetDescription()` doesn't seem to have a reference to the debugger, target or interpreter so I'm wondering what the next best options to determine if we're using colours is.

Reply:
This wouldn't be a problem if we make sure that whoever makes the Streams for use by the command interpreter consults GetUseColor when it creates the stream, and then we can rely on the stream to tell us what to do.  If you are calling stream API's that add the color for you, that should just work.  If you are doing a Print by hand with the color codes or not, you should consult the stream.  For that we'll have to add Stream::GetUseColor.  If we always get that right, the only problem will be if there are persistent streams, we may need to change their state or replace them when GetUseColor changes.

That seems the most principled way to solve this.

Reply:
I was speaking to Ismail about this and seems like it would be nice if we had a `Stream::GetUseColor` for this given that you can set whether or not the stream has colours at construction but you can't query the state of it after that.